transactor/src/main/scala/io/mediachain/types/CborSerialization.scala

```diff
@@ -1,51 +1,123 @@
-package io.mediachain.transactor
+package io.mediachain.types
 
 import io.mediachain.multihash.MultiHash
 
 
 object CborSerialization {
 
   import cats.data.Xor
-  import io.mediachain.transactor.Dummies.DummyReference
-  import io.mediachain.transactor.Types._
+  import io.mediachain.types.Datastore._
   import io.mediachain.util.cbor.CborAST._
   import io.mediachain.util.cbor.CborCodec
 
   import scala.util.Try
 
+  /**
+    * A mapping of `MediachainType` to the `CborDeserializer` to use when
+    * decoding objects of that type.
+    *
+    * Used to allow the transactors to deserialize chain cells into the most
+    * generic representation, while allowing other peers to deserialize into
+    * specific subtypes.
+    *
+    * Defaults to deserializing into specific subtypes; to override this
+    * behavior and deserialize into generic `EntityChainCell`s and
+    * `ArtefactChainCell`s, make an implicit `DeserializerMap` in the scope
+    * where you call the various `fromCbor` methods.
+    *
+    * e.g:
+    * ```
+    * implicit val deserializerMap = CborSerialization.transactorDeserializers
+    *
+    * val fooCell = fromCborBytes(fooCellBytes)
+    * ```
+    */
   type DeserializerMap = Map[MediachainType, CborDeserializer[CborSerializable]]
```

Comment:
ok so this impl is actually fine but i'm not sure that u have to pass around the implicit `derserializers` everywhere. it's really confusing to see it not explicitly passed in to all the fns that accept it as a parameter